### ISSUE TITLE
`overrides.rules.<rule>` does not override `rules.all`.

### ISSUE THREAD

**Issue by PhilMarsh** [astral-sh/ty#2961]
### Summary

Take this snippet which expectedly fails `invalid-return-type`:
```py
def foo() -> int:
    return None
```

These `pyproject.toml` configurations expectedly suppress the error:
- `rules.invalid-return-type` overriding `rules.all`:
  ```toml
  [tool.ty.rules]
  all = "error"
  invalid-return-type = "ignore"
  ```
- `overrides.rules.invalid-return-type` overriding `rules.invalid-return-type`:
  ```toml
  [tool.ty.rules]
  invalid-return-type = "error"
  
  [[tool.ty.overrides]]
  include = ["**"]
  
  [tool.ty.overrides.rules]
  invalid-return-type = "ignore"
  ```
- `overrides.rules.all` repeating `rules.all` _and_ `overrides.rules.invalid-return-type` overriding `overrides.rules.all`:
  ```toml
  [tool.ty.rules]
  all = "error"
  
  [[tool.ty.overrides]]
  include = ["**"]
  
  [tool.ty.overrides.rules]
  all = "error"
  invalid-return-type = "ignore"
  ```

But this configuration unexpectedly does **not** suppress the error:
- `overrides.rules.invalid-return-type` trying to override `rules.all` (without `overrides.rules.all`):
  ```toml
  [tool.ty.rules]
  all = "error"
  
  [[tool.ty.overrides]]
  include = ["**"]
  
  [tool.ty.overrides.rules]
  invalid-return-type = "ignore"
  ```

It seems like this last case should work the same is the rest.

### Version

ty 0.0.20 (2bd025243 2026-03-02)

**Comment by MichaReiser** [comment]
Thanks for the detailed write-up. I see where this comes from. I'm surprised that no existing test caught this. I'll take look

**Comment by PhilMarsh** [comment]
Thanks for fast fix!

I just found another issue with `all` that might just be another aspect of the same thing.

---

Given:
1. The same snippet which correctly fails `invalid-return-type`:
    ```py
    def foo() -> int:
        return None
    ```
2. `pyproject.toml` configuration that correctly suppresses the error:
    ```toml
    [tool.ty.rules]
    invalid-return-type = "ignore"
    ```

Running `ty check --error=invalid-return-type` expectedly overrides the config suppression and reports the error.

But running `ty check --error=all` does **not** override the config; the error remains suppressed.

---

From the description of https://github.com/astral-sh/ruff/pull/23712, it sounds like this _might_ be the same issue and flipping the precedence will fix it.
If it is the same, might be worth another test case?
If not, I can file a separate issue.

---

EDIT: This might be more related to https://github.com/astral-sh/ty/issues/2952 ? Using `--error=ALL` (all caps) _does_ override the config (and any subsequent `--ignore` flags).

**Comment by MichaReiser** [comment]
> But running ty check --error=all does not override the config; the error remains suppressed.

This is expected. CLI arguments take precedence over configuration files, even if the used selector is less specific.

**Comment by carljm** [comment]
> CLI arguments take precedence over configuration files, even if the used selector is less specific.

This seems to describe the opposite of the behavior @PhilMarsh reported:

> running `ty check --error=all` does **not** override the config; the error remains suppressed.

**Comment by MichaReiser** [comment]
Right. This now works as expected on https://github.com/astral-sh/ruff/pull/23712 (which is based on https://github.com/astral-sh/ty/issues/2952)
